sdk/storagev2_model.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2021 ProximaX Limited. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package sdk
+
+import (
+	"fmt"
+)
+
+type DataModificationState uint8
+
+const (
+	Succeeded DataModificationState = iota
+	Cancelled
+)
+
+type ActiveDataModification struct {
+	Id                 *Hash
+	Owner              *PublicAccount
+	DownloadDataCdi    *Hash
+	ExpectedUploadSize StorageSize
+	ActualUploadSize   StorageSize
+	FolderName         string
+	ReadyForApproval   bool
+}
+
+func (active *ActiveDataModification) String() string {
+	return fmt.Sprintf(
+		`
+			"Id": %s,
+			"Owner": %s,
+			"DownloadDataCdi": %s,
+			"ExpectedUploadSize": %d,
+			"ActualUploadSize": %d,
+			"FolderName": %s,
+			"ReadyForApproval": %t, 
+		`,
+		active.Id.String(),
+		active.Owner.String(),
+		active.DownloadDataCdi.String(),
+		active.ExpectedUploadSize,
+		active.ActualUploadSize,
+		active.FolderName,
+		active.ReadyForApproval,
+	)
+}
+
+type CompletedDataModification struct {
+	ActiveDataModification []*ActiveDataModification
```

Comment:
`ActiveDataModification` should  be embedded in `CompletedDataModification`

```
type CompletedDataModification struct {
	ActiveDataModification
	State                  DataModificationState
}
```

Reply:
I can change it but I'm unsure how to write the ActiveDataModification field in the DTOs for CompletedDataModification. Is there an example I can refer to?

Reply:
You can work with embedded structs like with common fields. One more question: why array `[]*ActiveDataModification`?

Reply:
I wasn't sure how to write the `ActiveDataModification` field in the DTOs for `CompletedDataModification` at first so I wrote it as an array.

I've updated it in 3431830 by referring to how the abstractTransactionDTO is written in ~/go-xpx-chain-sdk/sdk/transaction_internal.go.